common/src/main/scala/db/SqlRegistrationRepository.scala

```diff
@@ -87,18 +88,20 @@ class SqlRegistrationRepository[F[_]: Async](xa: Transactor[F])
       )
       ++ fr"GROUP BY token, platform"
       )
-      .query[(String, String)]
+      .query[(String, String, Option[String])]
       .stream
       .transact(xa)
-      .map{ case(token, platformString) => {
+      .map{ case (token, platformString, buildTierString) => {
         val maybePlatform = Platform.fromString(platformString)
         if(maybePlatform.isEmpty) {
           logger.error(s"Unknown platform in db $platformString")
         }
-        (token, maybePlatform)
+        val maybeBuildTier: Option[BuildTier] = BuildTier.fromString(buildTierString.toString)
```

Comment:
I'm rooting for you, but that's not quite what we want!
Have a look at what the [scala doc](https://www.scala-lang.org/api/current/scala/Option.html) define as "useful methods" here, and see if there's one that appear to have the type signature you need